### ISSUE TITLE
Convert to component integration test

### ISSUE THREAD

**Issue by Robdel12** [thefrontside/emberx-select#93]
This will probably end up happening when we get designs to make the demo page nicer. This will allow the demo page to be an actual demo and not actually our test bed.

**Comment by kategengler** [comment]
Some drive by thoughts here: For now, acceptance tests have much nicer capabilities and are as close to testing a real app using your addon as you can get, for that reason I prefer them for testing addons. However, I did find it awkward when PRing a fix to that bug that I was polluting the documentation just to be able to write a test. And, just now as I wrote this up, it occurred to me that the test bed could just be another route in the dummy app.

**Comment by Robdel12** [comment]
I agree, we _really_ like acceptance testing. The documentation part wasn't intended since it's truly just our test bed but it was added by @lydiaguarino. I have a feeling we're going to steal all that copy and move it to our nice example page once it's done and then the test bed would be hiding under some route like `sandbox-tests` or something like that.

**Comment by Robdel12** [comment]
[In the weekly email](https://github.com/thefrontside/oss-wumbo/blob/master/weekly-emails/july-5th-8th-2016.md) I mentioned:

> I’d like to have an offline discussion about whether it’s a good idea to build an application in our dummy apps as a test bed. I’ve discovered it requires some maintenance when making changes.

And I think this issue is related. When I started working on removing the two way data binding by default I discovered that the application we built as a test bed for x-select has maintenance baggage. 

I'm not arguing against building a dummy app for our test bed, but I would like to have a little discussion about it. I did like that I was feeling the effects of x-select changing in a way and that I had to go through and fix the broken application due to the changes made. There's something to be said about that. 

But on the flip side of things, we should consider the amount of effort it might take to keep the maintenance up? 

For my 0.2, I don't feel strong enough to make a decision. I still _really_ like acceptance testing our components. But I did want to raise the maintenance flag. 

CC: @cowboyd @flexyford / anyone else that would like to chime in.

**Comment by cowboyd** [comment]
I think we just need to acklowledge that we have demos that need to be maintained. As such, they should be tested so that we don't end up with breakage. But those demos are curated experiences and shouldn't need to be the go-to place  just because somebody is making a fix or adding a feature. After all, we're talking about rolling up all of these demos and putting them in a showcase, right?

I like @kategengler's suggestion: for cutting room floor acceptance tests, let's use a `/test-bed` route.

In summary:

Demos have their own tests which are changed only when changing the demos themselves. Addon tests happen as either an acceptance test in the testbed, or an integration test.

**Comment by Robdel12** [comment]
I'm going to close this since the ticket is no longer relevant.
